pil-analyzer/src/type_inference.rs

```diff
@@ -912,28 +985,51 @@ impl TypeChecker {
         &self,
         inferred_types: HashMap<String, Type>,
     ) -> Result<HashMap<String, HashMap<String, Type>>, Error> {
-        inferred_types.into_iter().map(|(name, inferred_type)| {
-            let (source_ref, declared_type) = self.declared_types[&name].clone();
-            let inferred_type = self.type_into_substituted(inferred_type.clone());
-            let inferred = self.to_type_scheme(inferred_type.clone());
-            let declared = declared_type.clone().simplify_type_vars();
-            if inferred != declared {
-                return Err(source_ref.with_error(format!(
-                    "Inferred type scheme for symbol {name} does not match the declared type.\nInferred: let{}\nDeclared: let{}",
-                    format_type_scheme_around_name(&name, &Some(inferred)),
-                    format_type_scheme_around_name(&name, &Some(declared_type),
-                ))));
-            }
-            let declared_type_vars = declared_type.ty.contained_type_vars();
-            let inferred_type_vars = inferred_type.contained_type_vars();
-            Ok((name.clone(),
-                inferred_type_vars
-                    .into_iter()
-                    .cloned()
-                    .zip(declared_type_vars.into_iter().map(|tv| Type::TypeVar(tv.clone())))
-                    .collect(),
-            ))
-        }).collect::<Result<_, Error>>()
+        inferred_types
+            .into_iter()
+            .map(|(name, inferred_type)| {
+                let declared_type = self.declared_types[&name].clone();
```

Comment:
Why not move this into `compare_type_schemes` as well? Maybe also name it `compare_inferred_type_to_declared`?